controllers/front/configuration.php

```diff
@@ -24,128 +24,21 @@
-                            : 'n/a',
-                        'php_version' => PHP_VERSION,
-                        'server_software' => $_SERVER['SERVER_SOFTWARE'],
-                        'server_name' => $_SERVER['SERVER_NAME'],
-                        'server_addr' => $_SERVER['SERVER_ADDR'],
-                        'database_version' => Db::getInstance()->getVersion(),
-                    ],
-                    'shop_configuration' => $config_manager->returnConfigurationOptions(),
-                ];
-
-                exit($this->setResponse(200, 'OK', $response));
-
-            case 'POST':
-            case 'PUT':
-                $json_data = Tools::file_get_contents('php://input');
-                $hash_algorithm = $this->getHashAlgorithm();
-                $hash_algos = array_intersect(array_merge(['sha3-256'], PHP_VERSION_ID < 70100 ? ['sha512'] : []), hash_algos());
-
-                if (in_array($hash_algorithm, $hash_algos, true)) {
-                    if (!hash_equals(hash($hash_algorithm, Api::getApiKey() . $json_data), $this->getSignature())) {
-                        exit($this->setResponse(400, 'Failed comparison of CR-Signature and shop hash.'));
-                    }
-                } else {
-                    exit($this->setResponse(403, 'Unsupported hash algorithm.'));
-                }
-
-                $configuration_options = json_decode($json_data, true);
-
-                if (is_array($configuration_options)) {
-                    $config_manager->updateConfiguration($configuration_options);
-
-                    exit($this->setResponse(204, ''));
-                }
-
-                exit($this->setResponse(400, 'Wrong input data.'));
-
-            default:
-                exit($this->setResponse(403, 'Access not allowed.'));
-        }
-    }
-
-    private function getSignature()
-    {
-        if (isset($_SERVER['HTTP_CR_SIGNATURE'])) {
-            return $_SERVER['HTTP_CR_SIGNATURE'];
-        }
-
-        return isset($_SERVER['HTTP_X_CR_SIGNATURE']) ? $_SERVER['HTTP_X_CR_SIGNATURE'] : '';
-    }
-
-    private function getHashAlgorithm()
-    {
-        if (isset($_SERVER['HTTP_CR_SIGNATURE_ALGO'])) {
-            return $_SERVER['HTTP_CR_SIGNATURE_ALGO'];
-        }
-
-        return isset($_SERVER['HTTP_X_CR_SIGNATURE_ALGO']) ? $_SERVER['HTTP_X_CR_SIGNATURE_ALGO'] : 'sha3-256';
-    }
-
-    /**
-     * @param $code
-     * @param $status_message
-     * @param array|null $data
-     * @return mixed
-     */
-    private function setResponse($code, $status_message, $data = null)
-    {
-        http_response_code($code);
-        header('Content-Type: application/json');
-
-        $response = ['status' => $status_message];
-
-        if ($data !== null) {
-            $response = array_merge($response, $data);
-        }
-
-        return json_encode($response);
+        exit(ApiService::processRequest('configuration'));
```

Comment:
`configuration` można by wrzucić do consta/enuma

Dlaczego tutaj jest `exit()`?

Reply:
To już opisałem poniżej. Pamiętaj, to świat wtyczek e-commerce, Presty, WordPress-ów i innych mutantów. Tu nie ma litości i KISS-ów, SOLID-ów i innych DRY-ów. ;-)

Reply:
Enuma nie mamy bo do niedawna trzymałem zgodność z PHP 5.6, a teraz z 7.1. Enum to dopiero PHP 8.1.